sbin/prepareWorkspace.sh

```diff
@@ -35,14 +34,55 @@ ALSA_LIB_CHECKSUM=${ALSA_LIB_CHECKSUM:-5f2cd274b272cae0d0d111e8a9e363f0878332915
 ALSA_LIB_GPGKEYID=${ALSA_LIB_GPGKEYID:-A6E59C91}
 FREETYPE_FONT_SHARED_OBJECT_FILENAME="libfreetype.so*"
 
+
+copyFromDir() {
+  echo "Copying OpenJDK source from  ${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to be built"
+  # we really do not want to use .git for dirs, as we expect user have them set up, ignoring them
+  local files=$(find "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}" -maxdepth 1 -mindepth 1 | grep -v -e "/workspace$" -e "/build$" -e "/.git" -e -"/build/")
+  cp -rf "$files" "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}/"
+}
+
+unpackFromArchive() {
+  echo "Extracting OpenJDK source tarbal ${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]} to $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} to be built"
+  # if the tarball contains .git files, they should be ignored later withut shame
+  # todo, support also zips?
+  pushd "./${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+    local topLevelItems=$(tar --exclude='*/*' -tf  "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}" | wc -l)
+    if [ "$topLevelItems" -eq "1" ] ; then
+      echo "Source tarball contains exaclty one directory, using"
+      tar --strip-components 1 -xf "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}"
+    else
+      echo "Source tarball does not contain a top level directory, using"
+      tar --strip-components 0 -xf "${BUILD_CONFIG[OPENJDK_FOREST_DIR_ABSPATH]}"
+    fi
+    rm -rf "build"
+  popd
+}
+
+copyFromDirOrUnpackFromArchive() {
+  echo "Removing copy of openjdk source repository of $(pwd)/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]} in 10 seconds..."
+  sleep 10
```

Comment:
Why have this delay?

Reply:
The `git reset` have same delay: https://github.com/adoptium/temurin-build/blob/646a79f2668378151985bc9517392ad955555ea6/sbin/prepareWorkspace.sh#L68

It may be good idea to wait a bit, before the finished build is accidently lost.